test/app_automate_driver_test.rb

```diff
@@ -0,0 +1,116 @@
+require 'test_helper'
+require_relative '../lib/features/support/app_automate_driver'
+require_relative '../lib/features/support/capabilities/devices'
+
+# Ensure at_exit blocks are eaten
+module Kernel
+  alias_method :old_at_exit, :at_exit
+  def at_exit
+  end
+end
+
+
+class AppAutomateDriverTest < Test::Unit::TestCase
+
+  USERNAME = "Username"
+  ACCESS_KEY = "Access_key"
+  LOCAL_ID = "Local_id"
+  APP_LOCATION = "App_location"
+  TEST_APP_URL = "Test_app_url"
+  TARGET_DEVICE = "ANDROID_9"
+  LOCAL_TUNNEL_COMMAND = "/BrowserStackLocal -d start --key #{ACCESS_KEY} --local-identifier #{LOCAL_ID} --force-local"
+
+  def test_defaults
+    AppAutomateDriver.any_instance.stubs(:upload_app).returns(TEST_APP_URL)
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID, TARGET_DEVICE, APP_LOCATION)
+
+    assert_equal(ACCESS_KEY, driver.instance_variable_get(:@access_key))
```

Comment:
Similar to https://github.com/bugsnag/maze-runner/pull/54#discussion_r284584573, shouldn't this be tested by other assertions instead? (in this particular case, checking `driver.server_url` and that username/access_key are used for uploading the app and starting the tunnel).

Similar comment applies to your next test method (`#test_overridden_location`)